### PR TITLE
Staticarrays

### PR DESCRIPTION
**Pull request by FedericoStra**
I modified `rRTNtoECI`, `rECItoRTN`, `sRTNtoECI` and `sECItoRTN` to make use of `SMatrix` and `SVector`. As a result, they are around 16x faster on my laptop.

I also did a general cleanup of the documentation, which contained several inconsistencies.

My original plan was to convert also the functions `bias_precession_nutation`, `earth_rotation`, `polar_motion`, `rECEFtoECI`, `rECItoECEF`, `sECEFtoECI` and`sECItoECEF`; but this is less trivial because the involved matrices are computed by `SOFA`, so the optimization should be implemented in that library.

### CODE REVIEW

**Comment by mykelk** [comment]
Awesome! Thanks for your contribution! @duncaneddy will do a review and then merge it in.

File: src/reference_systems.jl [hunk 1]
```diff
@@ -3,136 +3,216 @@ module ReferenceSystems
...
+using StaticArrays: SVector, SMatrix
+
 using SOFA
 using SatelliteDynamics.Constants
 using SatelliteDynamics.Universe: UT1_UTC, POLE_LOCATOR
 using SatelliteDynamics.Time: Epoch, mjd
 
+const idx123    = SVector(1, 2, 3)
+const idx456    = SVector(4, 5, 6)
+const idx123456 = SVector(1, 2, 3, 4, 5, 6)
```
Comment (duncaneddy) [inline src/reference_systems.jl:15]: These really should be placed at the top module level in `SatelliteDynamics.jl` and used consistently throughout all  submodule as to reduce reuse. But given that the StaticArray updates are confined to just this location, I'm fine leaving it here for now.

This is pedantic, but I would suggest renaming the variables `idx123` -> `idx1t3`, `idx456` -> `idx4t6` and `idx123456` -> `idx1t6` to make the variables consistent length and clearly ranges.

I would prefer not to have to use these const-defs at all, but I appreciate that they're necessary to preserve type stability and the performance improvements provided by StaticArrays, so I think the performance benefits in this case outweigh the readability concerns.

**Comment by duncaneddy** [comment]
@FedericoStra many thanks for your contribution!

I hadn't thought of using StaticArrays. That's great the update gives such a performance boost. I agree this update should be applied equally throughout the rest of the library as well as in SOFA.jl. There's a new SOFA release out, so SOFA.jl is due for an update anyways. I'll hopefully get to updating that package and the rest of the this one to use StaticArrays this week.

I left one minor comment, but it wasn't significant enough to delay the merge.

File: src/reference_systems.jl (hunk 1 above)
Comment (FedericoStra) [reply src/reference_systems.jl:15]: > This is pedantic, but I would suggest renaming the variables `idx123` -> `idx1t3`, `idx456` -> `idx4t6` and `idx123456` -> `idx1t6` to make the variables consistent length and clearly ranges.

I agree, I pushed a new revision with this change, if you want: https://github.com/sisl/SatelliteDynamics.jl/pull/3.

**Pull request merged**